### ISSUE TITLE
EventBridge pushes the message to the queue multiple time(s)

### ISSUE THREAD

**Issue by Salman-K** [localstack/localstack#3253]
EventBridge pushes the message to the queue multiple time(s)

**Type of request: Bug**

[x] bug report
[ ] feature request
Detailed description

When using eventsbridge on localstack with SQS as target, the message when matched through rules is being pushed several times to the queue.

**Expected behavior**
Message only appear once on the queue and is de-queued once read.

**Actual behavior**
Message appears on the queue several times, I counted at least 5 times before queue is empty.

Steps to reproduce

1. Setup localstack using docker
1. Create Queue
1. Create EventBus
1. Set Queue attributes to allow eventbridge access
1. Put target on eventbridge
1. Put new event onto eventbus
1. Check queue for message

**Command used to start LocalStack**

```
docker run -d -p "4566-4587:4566-4587" -p "4592-4593:4592-4593" -p 80:8080 -v /var/run/docker.sock:/var/run/docker.sock -e LAMBDA_EXECUTOR=docker,USE_LEGACY_PORTS=false --name aws localstack/localstack:latest
```

**Client code (AWS SDK code snippet, or sequence of "awslocal" commands)**

```
$AWS_REGION = 'us-east-1'
$LOCALSTACK_ENDPOINT = 'http://localhost'

aws --endpoint-url=${LOCALSTACK_ENDPOINT}:4566 sqs create-queue --queue-name orderTestQueue

aws --endpoint-url=${LOCALSTACK_ENDPOINT}:4566 --region ${AWS_REGION} events create-event-bus --name orderTestBus

aws --endpoint-url=${LOCALSTACK_ENDPOINT}:4566 --region ${AWS_REGION} events put-rule --name orderTestRuleMatchSource --event-bus-name orderTestBus --event-pattern file://rule-filter.json

aws --endpoint-url=${LOCALSTACK_ENDPOINT}:4566 sqs set-queue-attributes --queue-url "http://localhost:4566/queue/orderTestQueue" --attributes file://sqs-policy.json

aws --endpoint-url=${LOCALSTACK_ENDPOINT}:4566--region ${AWS_REGION} events put-targets --rule orderTestRuleMatchSource --event-bus-name orderTestBus --targets file://rule-queue-arn.json

aws --endpoint-url=${LOCALSTACK_ENDPOINT}:4566--region ${AWS_REGION} events put-events --entries file://order-data.json

aws --endpoint-url=${LOCALSTACK_ENDPOINT}:4566 sqs receive-message --queue-url "http://localstack:4566/queue/orderTestQueue"
```

**rule-filter.json**
{"source": [ "UpdateOrder" ]}

**order-data.json**
[{
"EventBusName": "orderTestBus",
"Source": "UpdateOrder",
"DetailType": "OrderAdded",
"Detail": "{"order_id":"a787ecb","quantity":"1","customer_id":"alpha"}"
}]

**sqs-policy.json**
{"Policy":"{"Version":"2012-10-17","Statement":[{"Effect":"Allow","Principal":{"AWS":"*"},"Action":"sqs:SendMessage","Resource":"arn:aws:sqs:us-east-1:000000000000:orderTestQueue","Condition":{"ArnEquals":{"aws:SourceArn":"arn:aws:events:us-east-1:000000000000:rule/orderTestRuleMatchSource"}}}]}"}

**rules-queue-arn.json**
[{"Id": "orderTestQueueCommand", "Arn": "arn:aws:sqs:us-east-1:000000000000:orderTestQueue", "InputPath": "$.detail"}]

When you push the following message
```
[{
"EventBusName": "orderTestBus",
"Source": "UpdateOrder",
"DetailType": "OrderAdded",
"Detail": "{"order_id":"a787ecb","quantity":"1","customer_id":"alpha"}"
}]
```

it appears on the queue atleast 5 times, even though it was pushed only once.

[rule-queue-arn.zip](https://github.com/localstack/localstack/files/5561437/rule-queue-arn.zip)

Referenced files are attached.
Many Thanks

┆Issue is synchronized with this [Jira Bug](https://localstack.atlassian.net/browse/LOC-330) by [Unito](https://www.unito.io/learn-more)

**Comment by macnev2013** [comment]
Thanks for reporting @Salman-K . We are not able to reproduce the issue.
Here's the steps we used to reproduce. Please help us understand if we are missing something, 
Also can you please provide us the localstack image version using command:  `docker images | grep localstack/localstack`

**$ AWS_REGION='us-east-1'
$ LOCALSTACK_ENDPOINT='http://localhost'**

**$ awslocal sqs create-queue --queue-name orderTestQueue**
```
{
    "QueueUrl": "http://localhost:4566/000000000000/orderTestQueue"
}
```

**$ awslocal events create-event-bus --name orderTestBus**
```
{
    "EventBusArn": "arn:aws:events:us-east-1:000000000000:event-bus/orderTestBus"
}
```

**$ awslocal  events put-rule --name orderTestRuleMatchSource --event-bus-name orderTestBus --event-pattern file://rule-filter.json**
```
{
    "RuleArn": "arn:aws:events:us-east-1:000000000000:rule/orderTestRuleMatchSource"
}
```

**$ awslocal  sqs set-queue-attributes --queue-url "http://localhost:4566/queue/orderTestQueue" --attributes file://sqs-policy.json**

**$ awslocal events put-targets --rule orderTestRuleMatchSource --event-bus-name orderTestBus --targets file://rule-queue-arn.json**
```
{
    "FailedEntryCount": 0,
    "FailedEntries": []
}
```

**$ awslocal events put-events --entries file://order-data.json**
```
{
    "Entries": [
        {
            "EventId": "ddf77595-dc30-4dcf-b3b6-318d3d54cab3"
        }
    ]
}
```

**$ awslocal  sqs receive-message --queue-url "http://localstack:4566/queue/orderTestQueue"**
```
{
    "Messages": [
        {
            "MessageId": "d7b3879e-341a-bdc9-7ee9-9b2a19b57c7d",
            "ReceiptHandle": "oynoivheopibhhhofdmwbaxtisgxqeemgrzxgjdajvvwokuyuvnhlmgiqiwimjgknxenwnfawiuuffoadlxcyvzbhclamaxuevdpihmrhumpcqlfgtdnhmqozrdzelzrjlbpzfwbyuyupmifkwzmqlqyhkvbfqaascthbdrjbzkbxfrhgjieuobuw",
            "MD5OfBody": "a03d83b3b928c21ab93603b315fd7ec5",
            "Body": "{\"order_id\": \"a787ecb\", \"quantity\": \"1\", \"customer_id\": \"alpha\"}",
            "Attributes": {
                "SenderId": "AIDAIT2UOQQY3AUEKVGXU",
                "SentTimestamp": "1606063135742",
                "ApproximateReceiveCount": "1",
                "ApproximateFirstReceiveTimestamp": "1606063143306"
            }
        }
    ]
}
```

**Files**

**order-data.json**
```
[{
	"EventBusName": "orderTestBus",
	"Source": "NewOrder",
	"DetailType": "OrderAdded",
	"Detail": "{\"order_id\":\"a787ecb\",\"quantity\":\"1\",\"customer_id\":\"alpha\"}"
}]
```

**rule-filter.json**
```
{
    "source": [ "NewOrder" ]
}
```

**rule-queue-arn.json**
```[{"Id": "orderTestQueueCommand", "Arn": "arn:aws:sqs:us-east-1:000000000000:orderTestQueue", "InputPath": "$.detail"}]```

**sqs-policy.json**
```{"Policy":"{\"Version\":\"2012-10-17\",\"Statement\":[{\"Effect\":\"Allow\",\"Principal\":{\"AWS\":\"*\"},\"Action\":\"sqs:SendMessage\",\"Resource\":\"arn:aws:sqs:us-east-1:000000000000:orderTestQueue\",\"Condition\":{\"ArnEquals\":{\"aws:SourceArn\":\"arn:aws:events:us-east-1:000000000000:rule/orderTestRuleMatchSource\"}}}]}"}```

**Comment by whummer** [comment]
Thanks for reporting @Salman-K @macnev2013 . This is a potential duplicate of #3254

----

Please note that this is the expected behavior - messages are by default **not** automatically deleted from the queue after receiving them.

Please refer to the AWS docs here : https://docs.aws.amazon.com/AWSSimpleQueueService/latest/SQSDeveloperGuide/sqs-visibility-timeout.html

> When a consumer receives and processes a message from a queue, the message remains in the queue

Also, you'll notice that the `ApproximateReceiveCount` increases, every time you receive the same message again.
```
ApproximateReceiveCount": "2",
...
ApproximateReceiveCount": "3",
```

Hope that helps - please let us know if there are any other issues. Thanks
